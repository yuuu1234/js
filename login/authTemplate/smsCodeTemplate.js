import React from 'react';
import { KeyboardAvoidingView } from 'react-native';
import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import styles from './style';

const withSMSCodeTemplate = ({
  Header, Main, Footer,
}) => {
  @observer
  class Screen extends React.Component {
    handleNavigation = (screenType) => {
      const { navigation } = this.props;
      navigation.navigate(screenType);
    }

    render() {
      return (
        <KeyboardAvoidingView behavior="padding" enabled style={styles.container}>
          { Header && <Header /> }
          { Main && <Main handleNavigation={this.handleNavigation} /> }
          { Footer && <Footer handleNavigation={this.handleNavigation} /> }
        </KeyboardAvoidingView>
      );
    }
  }

  Screen.propTypes = {
    navigation: PropTypes.shape({ navigate: PropTypes.func }).isRequired,
  };

  return Screen;
};

export { withSMSCodeTemplate };
