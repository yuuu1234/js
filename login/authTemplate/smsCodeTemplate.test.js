import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import PropTypes from 'prop-types';
import { withSMSCodeTemplate } from './smsCodeTemplate';

const MockHeader = () => <div testId="mock-header">Header</div>;
const MockMain = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-main">Main</div>;
const MockFooter = ({ handleNavigation }) => <div handleNavigation={handleNavigation} testId="mock-footer">Footer</div>;

MockMain.propTypes = { handleNavigation: PropTypes.func.isRequired };
MockFooter.propTypes = { handleNavigation: PropTypes.func.isRequired };

describe('withSMSCodeTemplate', () => {
  const params = {
    Header: MockHeader,
    Main: MockMain,
    Footer: MockFooter,
  };
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };
  const WithAuthComponent = withSMSCodeTemplate(params);
  const component = (
    <WithAuthComponent {...props} />
  );
  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call navigate correctly', () => {
    const main = wrapper.find({ testId: 'mock-main' });
    main.props().handleNavigation('mock screen');

    expect(props.navigation.navigate).toHaveBeenCalledWith('mock screen');
  });
});
