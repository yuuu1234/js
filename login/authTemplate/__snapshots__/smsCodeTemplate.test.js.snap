// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withSMSCodeTemplate should render component correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <div
    testId="mock-header"
  >
    Header
  </div>
  <div
    handleNavigation={[Function]}
    testId="mock-main"
  >
    Main
  </div>
  <div
    handleNavigation={[Function]}
    testId="mock-footer"
  >
    Footer
  </div>
</View>
`;
